activesupport/test/subscriber_test.rb

```diff
@@ -53,4 +68,26 @@ def test_does_not_attach_private_methods
 
     assert_equal [], TestSubscriber.events
   end
+
+  def test_deattaches_subscribers
+    TestSubscriber.deattach_from :doodle
+
+    sleep(60 * 2)
```

Comment:
Isn't there any way how to remove this sleep? Can't be `Concurrent::CountDownLatch` used for example?

Reply:
I am still debugging it. Main pull request is https://github.com/rails/rails/pull/35691

Test cases are failing in travis ci but passed in my local.
It just passed after adding `sleep`. Yeah I will try `Concurrent::CountDownLatch` as well. Thanks.

Reply:
Ahh, sorry. I'll move my comments to another PR.